### PR TITLE
2216 pass servers to exploiters

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #2216..

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] ~Was the CHANGELOG.md updated to reflect the changes?~
* [x] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running agent locally and by running unit tests
* [x] If applicable, add screenshots or log transcripts of the feature working

image

### CODE REVIEW

File: monkey/infection_monkey/exploit/HostExploiter.py [hunk 1]
```diff
@@ -58,6 +59,7 @@ def report_login_attempt(self, result, user, password="", lm_hash="", ntlm_hash=
     def exploit_host(
         self,
         host,
+        servers: List[str],
```
Comment (mssalvatore) [inline monkey/infection_monkey/exploit/HostExploiter.py:62]:
```suggestion
        servers: Sequence[str],
```

File: monkey/infection_monkey/exploit/exploiter_wrapper.py [hunk 2]
```diff
@@ -24,18 +24,21 @@ def __init__(
             telemetry_messenger: ITelemetryMessenger,
             event_queue: IEventQueue,
             agent_binary_repository: IAgentBinaryRepository,
+            servers: List[str],
```
Comment (mssalvatore) [inline monkey/infection_monkey/exploit/exploiter_wrapper.py:27]:
```suggestion
            servers: Sequence[str],
```

Comment (mssalvatore) [reply monkey/infection_monkey/exploit/exploiter_wrapper.py:27]: Servers need to be passed to the exploiter every time the exploiter is run. The list of servers may change over time.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

**Comment by mssalvatore** [comment]
How was this tested?

File: monkey/infection_monkey/master/exploiter.py [hunk 3]
```diff
@@ -53,6 +53,7 @@ def exploit_hosts(
         exploitation_config: ExploitationConfiguration,
         hosts_to_exploit: Queue,
         current_depth: int,
+        servers: Sequence[str],
```
Comment (VakarisZ) [inline monkey/infection_monkey/master/exploiter.py:56]: Maybe it's worth moving the servers to be object property? Most of Exploiter methods seem to use this parameter

Comment (VakarisZ) [reply monkey/infection_monkey/master/exploiter.py:56]: Perhaps not, but these method definitions are growing larger and larger. This class pass around a bunch of parameters it doesn't use and it doesn't need to know depth, servers or targets. It needs to know  only stop event and victim OS by the looks of it

**Review by VakarisZ — APPROVED** [review]

File: monkey/infection_monkey/master/exploiter.py (hunk 3 above)
Comment (mssalvatore) [reply monkey/infection_monkey/master/exploiter.py:56]: In the future, the list of severs may be updated with the list of all known servers from the island. The exploiter needs to be able to receive an updated list every time it runs.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**